File: frontend/src/models/building.ts
import {BuildingStatus} from "../enums/building-status";

export class Building {
id: number;
buildingAddress: string;
latitude: number;
longitude: number;
buildingStatus: BuildingStatus;

  constructor(
    id: number,
    buildingAddress: string,
    latitude: number,
    longitude: number,
    buildingStatus: BuildingStatus
  ) {
    this.id = id;
    this.buildingAddress = buildingAddress;
    this.latitude = latitude;
    this.longitude = longitude;
    this.buildingStatus = buildingStatus;
  }

}

File: demo/src/main/java/com/example/demo/entities/Building.java
package com.example.demo.entities;


import com.example.demo.enums.BuildingStatus;
import com.example.demo.enums.RoomStatus;
import jakarta.persistence.*;

import java.util.List;

@Entity
public class Building {
    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    private Long id;
    @Column(name = "address", unique = true)
    private String buildingAddress;

    @Column(name = "latitude")
    private Double latitude;

    @Column(name = "longitude")
    private Double longitude;

    public Long getId() {
        return id;
    }

    public String getBuildingAddress() {
        return buildingAddress;
    }

    public void setBuildingAddress(String buildingAddress) {
        this.buildingAddress = buildingAddress;
    }

    public Double getLongitude() {
        return longitude;
    }

    public void setLongitude(Double longitude) {
        this.longitude = longitude;
    }

    public Double getLatitude() {
        return latitude;
    }

    public void setLatitude(Double latitude) {
        this.latitude = latitude;
    }

    public BuildingStatus getBuildingStatus() {
        return buildingStatus;
    }

    public void setBuildingStatus(BuildingStatus buildingStatus) {
        this.buildingStatus = buildingStatus;
    }

    @Enumerated(value=EnumType.STRING)
    @Column(name = "status")
    private BuildingStatus buildingStatus;


}
